poetry/utils/env.py

```diff
@@ -1056,6 +1093,30 @@ def __init__(self, path: Path, base: Optional[Path] = None) -> None:
         self._path = path
         self._bin_dir = self._path / bin_dir
 
+        try:
+            python_executables = sorted(
+                [
+                    p.name
+                    for p in self._bin_dir.glob("python*")
+                    if re.match(r"python(?:\d+(?:\.\d+)?)?(?:\.exe)?$", p.name)
+                ]
+            )
+            self._executable = python_executables[0].rstrip(".exe")
```

Comment:
If I understand the code correct, it should work if there is only one python interpreter in the environment. But what happens if we have a python2, python3.6, python3.7? python2 will be selected. But we cannot be sure that this is the intended interpreter.

Can we make the following assumptions?:
* in a venv there will be only on interpreter
* in the system env there can be multiple interpreter

If so, we could look for python executables of the format `pythonX.Y`. If we receive exactly one, take it. If we get multiple, take the one with the same version as poetry is running.

BTW: `rstrip()` doesn't strip `.exe` from the end of the string. It removes any of the given character from the end until it hits a character not listed. A trap I run into regularly :smile:

Reply:
I updated the code to look for the executable of the child environment (if given). Otherwise the previous logic is used.

Reply:
Note that I intend to rewrite the environment management at some point but it will be after the release of the 1.2 version.